internal/graph/check_test.go

```diff
@@ -3412,3 +3418,365 @@ func TestCheckAssociatedObjects(t *testing.T) {
 		})
 	}
 }
+
+func TestConsumeUsersets(t *testing.T) {
+	model := parser.MustTransformDSLToProto(`
+				model
+					schema 1.1
+				type user
+				type group
+					relations
+						define member: [user]
+				type document
+					relations
+						define viewer: [group#member]`)
+	type dsResults struct {
+		tuples []*openfgav1.Tuple
+		err    error
+	}
+	tests := []struct {
+		name                         string
+		tuples                       []dsResults
+		usersetsChannelResult        []usersetsChannelStruct
+		ctxCancelled                 bool
+		expectedResolveCheckResponse *ResolveCheckResponse
+		errorExpected                error
+	}{
+		{
+			name: "userset_tuple_found",
+			tuples: []dsResults{
+				{
+					tuples: []*openfgav1.Tuple{
+						{Key: tuple.NewTupleKey("group:2", "member", "user:maria")},
+					},
+					err: nil,
+				},
+			},
+			usersetsChannelResult: []usersetsChannelStruct{
+				{
+					err:            nil,
+					objectRelation: "group#member",
+					objectIDs:      []string{"2"},
+				},
+			},
+			ctxCancelled: false,
+			expectedResolveCheckResponse: &ResolveCheckResponse{
+				Allowed: true,
+				ResolutionMetadata: &ResolveCheckResponseMetadata{
+					DatastoreQueryCount: 1,
+				},
+			},
+			errorExpected: nil,
+		},
+		{
+			name: "userset_tuple_found_multi_batches",
+			tuples: []dsResults{
+				// we expect 3 ds.ReadStartingWithUser to be called in response to 3 batches from usersetsChannel
```

Comment:
I don't follow this 🤔 When you send the first batch

```go
{
	err:            nil,
	objectRelation: "group#member",
	objectIDs:      []string{"1"},
},
```

we'll do `resp, err := checkAssociatedObjects(ctx, req, objectRel, objectIDs)`

which does this: `i, err := iteratorReadStartingFromUser(ctx, typesys, ds, req, objectRel, objectIDs)`

if you passed `1` as the objectId, why would the datastore return `{Key: tuple.NewTupleKey("group:11", "member", "user:maria")},`?

the `objectIDs` filter passed to `storage.ReadStartingWithUser` is a strict equality, not a prefix equality

Reply:
Ah, I understand what you are getting at.  Updated to reflect this.